consensus/istanbul/backend/backend.go

```diff
@@ -186,6 +186,21 @@ type Backend struct {
 	proxiedPeer consensus.Peer
 
 	newEpochCh chan struct{}
+
+	delegateSignFeed event.Feed
+}
+
+// This is for ethstats communication
```

Comment:
I notice that you have alot of checks similar to:

`if ...ProxiedPeer() != nil`

to figure out if this node is a proxy or is being proxied.

I think it will be more readable if you create two additional methods that are more explicit:

`IsProxy`
`IsProxiedValidator`

it will be easier to understand the code.